design/one-pager-package-signature-validation.md

```diff
@@ -0,0 +1,320 @@
+# Package Signature Validation
+
+* Owners: Jesse Sanford (@jessesanford), Dan Mangum (@hasheddan)
+* Reviewers: Crossplane Maintainers
+* Status: Draft
+
+## Background
+
+Crossplane packages (xpkgs) are the unit of extension in any Crossplane
+installation. Currently two package types are supported: Providers and
+Configurations. Providers may install Custom Resource Definitions (CRDs),
+Mutating Webhook Configurations, and Validating Webhook Configurations.
+They also may provide a (potentially self-referential) controller image
+reference, which Crossplane will install as a Deployment to reconcile the CRDs
+included in the package, which are commonly referred to as Managed Resources
+(MRs). Configurations may install Composite Resource Definitions (XRDs) and
+Compositions, which allow users to define abstractions that may compose any
+number of MRs or other Composite Resources (XRs).
+
+Crossplane packages are defined as OCI images, which allows Crossplane users to
+take advantage of existing tooling and distribution channels, as well as the
+myriad of benefits offered by a content-addressable API. Standardizing on this
+ecosystem also allows Crossplane to easily integrate with other projects that
+are built on top of the OCI specification.
+
+An important distinction between running an image on Kubernetes and installing a
+package via Crossplane is that the former invokes the container runtime on a
+Node in the cluster, while the latter fetches the image directly from the
+Crossplane process. Additionally, Crossplane maintains a separate cache of
+package content, allowing it to minimize the number of operations required when
+reconciling a package. Because package content is fetched directly, the
+Crossplane package manager is a convenient location to inject additional logic
+and safeguards. One such feature is image signature verification.
+
+## Goals
+
+- Enable provenance validation of crossplane packages
+- Allow the specification of a trusted public key to use for validation
```

Comment:
Sorry for the delay. Lots of things happening since early November. I would agree that we should consider allowing for keyless. The keyless isn't much harder to do now. There is some prior art in this PR from @developer-guy on flux here: https://github.com/fluxcd/source-controller/pull/876/files

Reply:
@stevehipwell I believe that we can now do a per-package set of secrets for validation with #3552 let me know what you think.